akka-docs/src/main/paradox/typed/persistence.md

```diff
@@ -48,6 +48,14 @@ The components that make up a EventSourcedBehavior are:
 * `commandHandler` defines how to handle command by producing Effects e.g. persisting events, stopping the persistent actor.
 * `eventHandler` returns the new state given the current state when an event has been persisted.
 
+@@@ div { .group-java }
+
+Note that the concrete class does not contain any fields with state like a regular POJO. All state of the 
+`EventSourcedBehavior` must be represented in the `State` or else they will not be persisted and therefore be
+lost when the actor is stopped or restarted.
```

Comment:
Maybe add: Updates to the `State` are always performed in the `eventHandler` based on the events.